documentation/openapi-authoring-automated-guidelines.md

```diff
@@ -2581,4 +2583,307 @@ The following would be valid:
   }
 ...
 ```
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [ARM](#arm-violations): [Errors](#arm-errors) or [Warnings](#arm-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="r4014" ></a>R4014 AllResourcesMustHaveGetOperation
+
+**Category** : ARM Warning
```

Comment:
As this Rule is mainly to detect resource model which don't path/get operation.  it is hard to know whether it is a really resource type from model, so we put it a warning first.  @ ARM to confirm

Reply:
What do you want me to confirm?  If there is a PUT or a LIST on a URI there should be an equivalent GET for individual resources at that same path (i.e. GET on the PUT's URI, or GET on the LIST + {name} URI).  This doesn't seem like it should be driven off the definitions, it should be driven off the paths in the spec.

Reply:
Thanks @pilor, yes we want to confirm the approach to  distinguish resourceType in path.  your idea looks  good.
but Is there any exception? e.g. /operations.  if no ideal solution, we may put this rule as warn, not error initially

Reply:
/operations should be the only legitimate exception, good call on that.  I'm sure there are plenty of places where RPs have done this but it should be something that requires a suppression in order to use since it goes against the RPC.  Is there a way to run the rule against all existing specs to see what the current problem space is?

Reply:
Actually we already scan the whole specs, and over 300 violations of this rule, a lot of them only have list model operation and  no put operation and point get  operation , so  it's hard to say that these models  are ARM resource, as they can be  some kinds of operations' result